atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -807,17 +807,11 @@ private static LockAndTimestampServices getLockAndTimestampServices(
             MetricsManager metricsManager,
             Supplier<ServerListConfig> timelockServerListConfig,
             String userAgent) {
-        LockService lockService = new ServiceCreator<>(metricsManager, LockService.class, userAgent)
-                .applyDynamic(timelockServerListConfig);
-
-        TimelockRpcClient timelockRpcClient = new ServiceCreator<>(metricsManager, TimelockRpcClient.class, userAgent)
-                .applyDynamic(timelockServerListConfig);
-
-        TimelockService timelockService = LeasingTimelockClient.create(timelockRpcClient);
-
-        TimestampManagementService timestampManagementService =
-                new ServiceCreator<>(metricsManager, TimestampManagementService.class, userAgent)
-                        .applyDynamic(timelockServerListConfig);
+        ServiceCreator creator = ServiceCreator.withPayloadLimiter(metricsManager, userAgent, timelockServerListConfig);
```

Comment:
Why do we apply the limiter here but not on the other similar calls in this class? I assume this is how we achieve the difference in behavior between embedded and external lock services, but `TransactionManagers` is hard enough to read that this isn't clear.

For that matter, do we even want the behavior difference for embedded? Normalizing would make this a break for embedded, but it would be unfortunate to design an application that worked until you moved to external Timelock.

Reply:
Discussed offline. It would be nice to unify behavior, but we don't want to push the break.

Raw services = embedded, so we don't apply the limit there.